framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1238,43 +1262,43 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
 
     /**
      * Checks that the following rule is satisfied: The type on a constructor declaration must be a
-     * supertype of the return type of "super()" invocation within that constructor (except for the
-     * Initialization and Rawness Checkers).
+     * supertype of the return type of "this()" invocation within that constructor.
      */
-    void checkSuperConstructorCall(MethodInvocationTree node) {
-        if (!TreeUtils.isSuperCall(node)) {
-            return;
-        }
-        TreePath path = atypeFactory.getPath(node);
+    protected void checkThisConstructorCall(MethodInvocationTree thisCall) {
```

Comment:
We could make this method `final`, to not allow overrides in subclasses.
At least, there should be a reference to `checkThisOrSuperConstructorCall` to allow easy adaptation of behavior.